client/components/app-promo/test/index.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { createStore } from 'redux';
+import { Provider } from 'react-redux';
+import { expect } from 'chai';
+import { mount } from 'enzyme';
+import { spy, stub } from 'sinon';
+import mockery from 'mockery';
+
+import useFakeDom from 'test/helpers/use-fake-dom';
+import useMockery from 'test/helpers/use-mockery';
+
+describe( 'AppPromo', ( ) => {
+	let AppPromo;
+	const store = createStore( stub() );
+
+	useFakeDom();
+	useMockery();
+
+	const desktopPromoKey = 'desktop_promo_dismissed';
+	let desktopPromoDismissed = false;
+	let localeSlug = 'en';
+	let isMobile = false;
+	let isChromeOS = false;
+
+	const storeMock = {
+		get: ( val ) => {
+			if ( val === desktopPromoKey ) {
+				return desktopPromoDismissed;
+			}
+
+			return undefined;
+		},
+		set: spy()
+	};
+
+	const userUtilsMock = { getLocaleSlug: ( ) => localeSlug };
+	const viewportMock = { isMobile: ( ) => isMobile };
+	const userAgentUtilsMock = { isChromeOS: ( ) => isChromeOS };
+	const appPromoDetails = {
+		promo_code: 'a0001',
+		message: 'WordPress.com your way  — desktop app now available for Mac, Windows, and Linux.'
+	};
+	const appPromoLink = 'http://www.wordpress.com';
+	const promoRetrieverMock = {
+		getRandomPromo: ( ) => appPromoDetails,
+		getPromoLink: ( ) => appPromoLink
+	};
+
+	const appStoreComponent = ( ) => {
+		return (
+			<Provider store={ store } >
+				<AppPromo location="reader" />
+			</Provider>
+			);
+	};
```

Comment:
I think that if you drop the function part of this you can just assign the React component itself

```js
const AppWithStore = (
	<Provider store={ store }>
		<AppPromo location="reader" />
	</Provider>
);
```

Then below you can just `mount( AppWithStore )`